File: solutions/java/bob/1/src/main/java/Bob.java
class Bob {

    public String hey(String say) {
        var s = say.trim();

        if (s.isBlank()) {
            return "Fine. Be that way!";
        }

        boolean isQuestion = s.endsWith("?");
        boolean isYell = s.toUpperCase().equals(s)
                && s.chars().filter(Character::isLetter)
                .findAny()
                .isPresent();

        if (isYell && isQuestion) {
            return "Calm down, I know what I'm doing!";
        } else if (isQuestion) {
            return "Sure.";
        } else if (isYell) {
            return "Whoa, chill out!";
        } else {
            return "Whatever.";
        }
    }
}

File: solutions/typescript/bob/1/bob.ts
export function hey(message: string): string {
  const say = message.replace(/\s/g, "")
  if (say === "") return "Fine. Be that way!"
  
  const isQuestion = say.endsWith("?")
  const isYellilng = message.toUpperCase() === message && /[a-zA-Z]/.test(message)

  if (isYellilng && isQuestion) return "Calm down, I know what I'm doing!"
  if (isQuestion) return "Sure."
  if (isYellilng) return "Whoa, chill out!"
  
  return "Whatever."
}
